diff --git a/tools/releasetools/ota_from_target_files.py b/tools/releasetools/ota_from_target_files.py
index 4d33c57..2dcf983 100755
--- a/tools/releasetools/ota_from_target_files.py
+++ b/tools/releasetools/ota_from_target_files.py
@@ -720,6 +720,15 @@ esac''')
     script.Print('*** Restoring Theme    ***')
     script.AppendExtra('run_program("/sbin/sh", "/tmp/overlay.sh", "restore");')
 
+  # gapps-config
+  script.Print('*** gapps-config.txt   ***')
+  common.ZipWriteStr(output_zip, "create-gapps-config.sh", '''#!/sbin/sh
+mkdir -p /persist
+echo PackageInstallerGoogle > /persist/gapps-config.txt
+''')
+  script.AppendExtra('package_extract_file("create-gapps-config.sh", "/tmp/create-gapps-config.sh");')
+  script.AppendExtra('run_program("/sbin/sh", "/tmp/create-gapps-config.sh");')
+
   if not OPTIONS.no_supersu_system or not OPTIONS.no_preserve_themes:
     # Unmount /system
     script.Unmount("/system")
