diff --git a/libs/hwui/Android.mk b/libs/hwui/Android.mk
index 366ef71..3ecd045 100644
--- a/libs/hwui/Android.mk
+++ b/libs/hwui/Android.mk
@@ -141,7 +141,7 @@ endif
 
 ifdef HWUI_COMPILE_FOR_PERF
     # TODO: Non-arm?
-    hwui_cflags += -fno-omit-frame-pointer -marm -mapcs
+    hwui_cflags += -fno-omit-frame-pointer -marm 
 endif
 
 # This has to be lazy-resolved because it depends on the LOCAL_MODULE_CLASS
